indigo/docs/guides/howto-hello-indigo-tyrian.md

```diff
@@ -0,0 +1,418 @@
++    case tyrianSubSystem.TyrianEvent.Receive(msg) =>
++      val pt               = Point(100, 100)
+       val adjustedPosition = pt - model.center
+```
+
+We've also added a line below that to set a `pt` variable as we now longer have
+a mouse position. For now, we've set that to 100,100 like so which is just a
+fixed point for the dots to rotate around.
+
+Once more build the project in mill or sbt and then run `yarn start`. This time
+you'll notice that clicking the game doesn't do anything, but clicking the
+button adds a new rotating dot.
+
+We've now successfully gotten Tyrian to talk to Indigo. In the next part we'll
+be doing the opposite, and adding a counter for the number of dots.
+
+## Indigo to Tyrian Communication
+
+In this part we're going to store a count of the number of dots that Indigo is
+currently displaying. To do this, first update the model in `HelloTyrian.scala`
+so that it has a `count` property which is initialised to zero, like so:
+
+```diff
+-final case class TyrianModel(bridge: TyrianIndigoBridge[Int])
++final case class TyrianModel(bridge: TyrianIndigoBridge[Int], count: Int)
+object TyrianModel:
+-  val init: TyrianModel
++  val init: TyrianModel =
++    TyrianModel(TyrianIndigoBridge(), 0)
+```
+
+We'll need to display that count on the website. To do this, we simply modify our
+counting `div` to display what's in the `model` like so
+
+```diff
+-      div(`class` := "counter")(),
++      div(``class`` := "counter")(model.count.toString)
+```
+
+A new message type is also required as before, so we'll add a new one like so:
+
+```diff
+   enum Msg:
+-    case StartIndigo extends Msg
++    case StartIndigo             extends Msg
+-    case AddDot      extends Msg
++    case AddDot                  extends Msg
++    case IndigoReceive(msg: Int) extends Msg
+```
+
+We'll need to subscribe to incoming Indigo messages so that we can act on them.
+This can be done using the bridge subscriptions like so:
+
+```diff
+   def subscriptions(model: TyrianModel): Sub[Msg] =
+-    Sub.Empty
++    model.bridge.subscribe { case msg =>
++      Some(Msg.IndigoReceive(msg))
++    }
+```
+
+The final part on the Tyrian side is then to update the model once we get a
+message from Indigo. Indigo will send an Integer message back to Tyrian, which
+we can use directly in our model To do this we simply add the following to our
+`update` pattern:
+
+```diff
+     msg match
+       case Msg.AddDot =>
+         (model, model.bridge.publish(IndigoGameId(gameDivId), 0))
++      case Msg.IndigoReceive(msg) =>
++        (model.copy(count = msg), Cmd.Empty)
+       case Msg.StartIndigo =>
+```
+
+For the Indigo side we need to modify the `updateModel` so that the new model is
+assigned to a variable for later use. We can then add a global event through
+the Tyrian Subsystem letting the website know how many dots we have. This is done
+with the following:
+
+```diff
+     case tyrianSubSystem.TyrianEvent.Receive(msg) =>
```

Comment:
`msg` is never used, which is a shame. Could you send over the x and y for the point or something?

Reply:
Maybe, but ultimately it's the page X and Y, it's just simpler I think to ignore it?